File: Semester4/Systems for Design and Implementation/mpp-proiect-csharp-raulfrandes/model/validator/ParticipantValidator.cs
namespace model.validator
{
    public class ParticipantValidator : IValidator<Participant>
    {
        public void Validate(Participant entity)
        {
            string errorMess = "";
            if (string.IsNullOrEmpty(entity.Name))
            {
                errorMess += "Invalid name - cannot be null\n";
            }

            int codeI = int.Parse(entity.Code);
            if (codeI < 1000 || codeI > 9999)
            {
                errorMess += "Invalid code - must be between 1000 or 9999";
            }
            if (!string.IsNullOrEmpty(errorMess))
            {
                throw new ValidatorException(errorMess);
            }
        }
    }
}

File: Semester4/Systems for Design and Implementation/mpp-proiect-java-raulfrandes/TriathlonModel/src/main/java/triathlon/model/validator/ParticipantValidator.java
package triathlon.model.validator;


import triathlon.model.Participant;

public class ParticipantValidator implements Validator<Participant> {
    @Override
    public void validate(Participant entity) throws ValidatorException {
        String errorMess = "";
        if (entity.getName().isEmpty()) {
            errorMess += "Invalid name - cannot be null";
        }
        int codeI = Integer.parseInt(entity.getCode());
        if (codeI < 1000 || codeI > 9999) {
            errorMess += "Invalid code - must be an number between 1000 and 9999";
        }
        if (!errorMess.isEmpty()) {
            throw new ValidatorException(errorMess);
        }
    }
}
